src/services/core.service.ts

```diff
@@ -184,6 +198,35 @@ export class CoreService {
             }).catch((err: any) => { this.logger.error('Load translations failed.', err.message); });
     }
 
+    public loadTranslationsAndPattern(language: string, region: string): Promise<any> | Promise<any>[] {
+        const cachePattern = this.cacheManager.lookforPattern(language, region);
+        if (typeof cachePattern !== 'undefined') {
+            const cacheTranslation = this.cacheManager.lookforTranslationByComponent(this.config.component, language);
+            if (typeof cacheTranslation === 'undefined') {
+                return Promise.resolve({
+                    components: [cacheTranslation],
+                    pattern: cachePattern
+                });
+            }
+        }
```

Comment:
1.  If this data can be checked in advance to keep loadTranslationsAndPattern clean.
2. The translation and pattern data validation should not be nested.

Reply:
Alreay modifed